### PR TITLE
Added Google Drive functions

### PR DESCRIPTION
**Pull request by dd-gamer**
Instruction how to set up will follow.

### CODE REVIEW

File: data/config-smple.yml [hunk 1]
```diff
@@ -5,7 +5,9 @@ blacklist: [
 ]
 courseConfigs: { }
 downloadFolder: "Files"
+drive_root_folder:
```
Comment (MrDrache333) [inline data/config-smple.yml]: Maybe put all Google-Drive related settings into another object like
```yaml
googleDrive:
  enabled: false
  rootDirectory: "Something"
  ```
  Also are the credentials stored inside "data/credentials" or outside? Maybe put it inside the data Folder inside its own folder like "googleDrive" or something?

File: src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/config/Config.java [hunk 2]
```diff
@@ -13,6 +13,8 @@ public class Config {
     private String api_endpoint;
     private String api_username;
     private String api_password;
+    private boolean use_google_drive;
```
Comment (MrDrache333) [inline src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/config/Config.java]: See data/config-smple.yml

File: src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/googleDrive/driveFunctions.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
+package de.oelrichsgarcia.studipTelegramBot.studipTelegramBot.googleDrive;
```
Comment (MrDrache333) [inline src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/googleDrive/driveFunctions.java]: Would be nice if you add some Comments and JavaDoc. Maybe even some unit-Tests if you can

File: src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/studip/StudIPBot.java [hunk 4]
```diff
@@ -236,11 +237,17 @@ private void fetchFilesForFolder(StudIPFolder folder) throws MalformedURLExcepti
      * @param objects  the objects
      * @param basePath the base path
      */
-    public void downloadFilesAndCreateFolders(ArrayList<StudIPObject> objects, Path basePath) {
+
+    public void downloadFilesAndCreateFolders(ArrayList<StudIPObject> objects, Path basePath, boolean useGoogleDrive, String drive_root_folder) {
+        checkAndCreateFolder(basePath);
```
Comment (MrDrache333) [inline src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/studip/StudIPBot.java:242]: Has been moved to Downloader

File: src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/studip/StudIPBot.java [hunk 5]
```diff
@@ -257,12 +264,18 @@ public void downloadFilesAndCreateFolders(ArrayList<StudIPObject> objects, Path
                     if (object.getUpdated().getTime() > file.lastModified()) {
                         Sout("Updating outdated file " + basePath + "/" + object.getName() + " " + fileSize);
                         downloadFile((StudIPFile) object, basePath);
+                        if(useGoogleDrive) {
```
Comment (MrDrache333) [inline src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/studip/StudIPBot.java:267]: Please use a custom Downloader implemnents Downloader

File: src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/studip/StudIPBot.java [hunk 6]
```diff
@@ -257,12 +264,18 @@ public void downloadFilesAndCreateFolders(ArrayList<StudIPObject> objects, Path
...
+                            uploadToGoogleDrive((StudIPFile) object, basePath, drive_root_folder);
+                        }
                         updateSummary.addUpdatedFile((StudIPFile) object);
 
                     }
                 } else {
                     Sout("Downloading new file " + basePath + "/" + object.getName() + " " + fileSize);
                     downloadFile((StudIPFile) object, basePath);
+                    if(useGoogleDrive) {
```
Comment (MrDrache333) [inline src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/studip/StudIPBot.java:276]: Please load the GoogleDriveDownloader during config loading so the StandardDownloader gets replaced by yours

File: src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/studip/StudIPBot.java [hunk 7]
```diff
@@ -295,6 +308,63 @@ private void downloadFile(StudIPFile file, Path path) {
         }
     }
 
+    private void uploadToGoogleDrive(StudIPFile file, Path path, String drive_root_folder) {
```
Comment (MrDrache333) [inline src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/studip/StudIPBot.java:311]: Please move to GoogleDriveDownloader

**Review by MrDrache333 — CHANGES_REQUESTED** [review]
Please move your Implementation to GoogleDriveDownloader located at de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/studip/api/download
Please add some documentation.

File: src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/config/Config.java (hunk 2 above)
Comment (dd-gamer) [reply src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/config/Config.java]: done

File: src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/googleDrive/driveFunctions.java (hunk 3 above)
Comment (dd-gamer) [reply src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/googleDrive/driveFunctions.java]: Done

File: src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/studip/StudIPBot.java (hunk 5 above)
Comment (dd-gamer) [reply src/main/java/de/oelrichsgarcia/studipTelegramBot/studipTelegramBot/studip/StudIPBot.java:267]: Can't do that, cause it builds on top of the downloads. Files need to exist locally, before they can be uploaded,

**Review by MrDrache333 — APPROVED** [review]

**Pull request merged**